osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/stub/DlmsConnectionStub.java

```diff
@@ -131,8 +161,33 @@ public void clearRequestedAttributeAddresses() {
     this.requestedAttributeAddresses = new ArrayList<>();
   }
 
+  public List<SetParameter> getSetParameters(final AttributeClass attributeClass) {
+    return this.parametersSet.stream()
+        .filter(
+            setParameter ->
+                setParameter.getAttributeAddress().getId() == attributeClass.attributeId())
+        .collect(Collectors.toList());
+  }
+
+  public boolean hasMethodBeenInvoked(final MethodClass methodClass) {
+    return this.methodsInvoked.stream()
+        .anyMatch(parameter -> parameter.getId() == methodClass.getMethodId());
+  }
+
+  public void addReturnValue(
+      final AttributeAddress attributeAddress,
+      final DataObject dataObject,
+      final int nrOfInvocations) {
+    for (int i = 0; i < nrOfInvocations; i++) {
+      if (!this.returnValues.containsKey(this.getKey(attributeAddress))) {
+        this.returnValues.put(this.getKey(attributeAddress), new ArrayList<>());
+      }
+      this.returnValues.get(this.getKey(attributeAddress)).add(dataObject);
+    }
```

Comment:
This piece of code puzzles me: For the number of invocations you want to add the same dataObject to the same address? At least you could put the if-statement outside the for me

Reply:
I changed it into more readable solution. I just adds an empty list if no value is present